scripts/src/client.ts

```diff
@@ -62,27 +58,44 @@ export class ClientError extends Error {
 }
 
 export class ClientRequests {
-	public static async create(data: any, headers?: StringMap) {
-		const res = await axios.post<{ auth: AuthToken; }>(MARKETPLACE_BASE + "/v2/users", data, { headers });
-		return new ClientRequests(res.data.auth);
+	public static async create(data: SignInPayload, headers?: StringMap) {
+		// get blockchain version for current app
+		const appId = this.extractAppId(data.jwt);
+		const blockchainVersion = (await axios.get<number>(
+				MARKETPLACE_BASE + `/v2/applications/${ appId }/blockchain_version`)
+		).data.toString() as BlockchainVersion;
```

Comment:
These 3 lines are butt ugly, can you extact the url string into a const please

Reply:
I don't see why extracting the URL to a consts makes it nicer. The rest of the code uses inline urls when making requests. The second part of `.data.toString()` can be moved/ turned into a function maybe